pkg/proxy/ipvs/proxier.go

```diff
@@ -1350,17 +1353,22 @@ func (proxier *Proxier) syncEndpoint(svcPortName proxy.ServicePortName, onlyNode
 }
 
 func (proxier *Proxier) cleanLegacyService(atciveServices map[string]bool, currentServices map[string]*utilipvs.VirtualServer) {
+	unbindIPAddr := sets.NewString()
 	for cS := range currentServices {
 		if !atciveServices[cS] {
 			svc := currentServices[cS]
 			err := proxier.ipvs.DeleteVirtualServer(svc)
 			if err != nil {
 				glog.Errorf("Failed to delete service, error: %v", err)
 			}
-			err = proxier.ipvs.UnbindVirtualServerAddress(svc, DefaultDummyDevice)
-			if err != nil {
-				glog.Errorf("Failed to unbind service from dummy interface, error: %v", err)
-			}
+			unbindIPAddr.Insert(svc.Address.String())
+		}
+	}
+
+	for _, addr := range unbindIPAddr.UnsortedList() {
+		err := utilproxy.UnbindAddress(addr, DefaultDummyDevice, proxier.netlinkHandle)
+		if err != nil {
+			glog.Errorf("Failed to unbind service from dummy interface, error: %v", err)
```

Comment:
nits:

service -> IP address?

output IP address and dummy interface name?

Reply:
The IP address and interface name can be printed in the `err` returned by `UnbindAddress`.

Reply:
If so, I would prefer directly return error in `UnbindAddress()` and then log detailed message here.